cpp/src/io/parquet/writer_impl.cu

```diff
@@ -85,6 +87,49 @@ parquet::Compression to_parquet_compression(compression_type compression)
   }
 }
 
+size_type column_size(column_view const& column, rmm::cuda_stream_view stream)
+{
+  if (column.num_children() == 0) {
+    return is_fixed_width(column.type()) ? size_of(column.type()) * column.size() : 0;
+  }
+
+  if (column.type().id() == type_id::STRING) {
+    auto const scol = strings_column_view(column);
+    return cudf::detail::get_value<size_type>(scol.offsets(), column.size(), stream) -
+           cudf::detail::get_value<size_type>(scol.offsets(), 0, stream);
+  } else if (column.type().id() == type_id::STRUCT) {
+    auto const scol = structs_column_view(column);
+    size_type ret   = 0;
+    for (int i = 0; i < scol.num_children(); i++) {
+      ret += column_size(scol.get_sliced_child(i), stream);
+    }
+    return ret;
+  } else if (column.type().id() == type_id::LIST) {
+    auto const lcol = lists_column_view(column);
+    return column_size(lcol.get_sliced_child(stream), stream);
+  }
+
+  return 0;
+}
+
+// checks to see if the given column has a fixed size.  This doesn't
+// check every row, so assumes string and list columns are not fixed, even
+// if each row is the same width.
+// TODO: update this if FIXED_LEN_BYTE_ARRAY is ever supported for writes.
+bool is_col_fixed_width(column_view const& column)
```

Comment:
Does this refer to Parquet columns in the output file?

Reply:
Well, these are `parquet_column_views` of the input table.  I had thought that at this level (schema leaf nodes), that the data types would be down to either fixed types, or lists/strings.  I'm still not quite sure why structs make it this far.

Reply:
yeah, looking at the FIXED_LEN_BYTE_ARRAY comment, it seems like the logic applies to Parquet columns/types.

Reply:
The FIXED_LEN_BYTE_ARRAY comment is in reference to #12590.  I'm assuming that once cuDF can read Parquet FLBA's, we'll want to write them too. IIRC right now FLBA is only used to encode decimals, so the cuDF column type here will be fixed len.  But if Parquet FLBA's turn into cuDF list(byte) or some such, then we'll need some schema info here to figure out whether it's fixed len or variable.